server/src/main/java/org/elasticsearch/transport/RemoteClusterService.java

```diff
@@ -95,6 +95,12 @@
         (ns, key) -> boolSetting(key, TransportSettings.TRANSPORT_COMPRESS,
             new RemoteConnectionEnabled<>(ns, key), Setting.Property.Dynamic, Setting.Property.NodeScope));
 
+    public static final Setting.AffixSetting<Boolean> REMOTE_CLUSTER_COMPRESS_RAW_DATA = Setting.affixKeySetting(
+        "cluster.remote.",
+        "transport.compress_raw_data",
```

Comment:
I would prefer `compress_indexing_data`. I think that can also meaningfully cover replaying operations and CCR indexing. Other suggestions are most welcome though.